subprojects/core/src/main/java/org/gradle/api/internal/file/FileOrUriNotationConverter.java

```diff
@@ -98,24 +100,41 @@ public void convert(Object notation, NotationConvertResult<? super Object> resul
                 result.converted(new File(uriDecode(notationString.substring(5))));
                 return;
             }
+            if (notationString.startsWith("http:") || notationString.startsWith("https:")) {
+                convertToUrl(notationString, result);
+                return;
+            }
             Matcher schemeMatcher = URI_SCHEME.matcher(notationString);
             if (schemeMatcher.matches()) {
                 String scheme = schemeMatcher.group(1);
-                if (new File(scheme).isDirectory()) {
+                if (isWindowsRootDirectory(scheme)) {
                     result.converted(new File(notationString));
                     return;
                 }
-                try {
-                    result.converted(new URI(notationString));
-                    return;
-                } catch (URISyntaxException e) {
-                    throw new UncheckedIOException(e);
-                }
+                convertToUrl(notationString, result);
+                return;
             }
             result.converted(new File(notationString));
         }
     }
 
+    private boolean isWindowsRootDirectory(String scheme) {
+        if (OperatingSystem.current().isWindows()) {
+            java.nio.file.FileSystem fs = FileSystems.getDefault();
+            Path path = fs.getPath(scheme);
```

Comment:
This is the same as `Path path = Paths.get(scheme)`, isn't it?

Reply:
Right, that's what happens when your code base is mostly on Java 6 =P